### PR TITLE
feat(accordion): add align prop to component and skeleton to change heading alignment

### LINKED ISSUE #4584: [Accordion] add `align` prop to align chevron to "start" of heading
## Summary

We recently wrapped Carbon's `Accordion` component to add an `align` prop. This prop allows user's to decide whether the `Accordion` chevron should be aligned at the start of the heading (which is *our* default) or at the end of the heading (which is Carbon's default).

So I was wondering if you'd like for me to contribute this upstream to `carbon-components-react`. The one difference would be, whereas our default is `start` alignment, yours would be `end` just due to the current state of your `Accordion`.

### Example of `align='start'`:

Screen Shot 2019-11-06 at 9 33 04 AM

## Justification

There are instances in the IBM Security portfolio where we need either `start` or `end` alignment.

`start` is our default because from the get-go, we needed to deviate a bit from the current Carbon `Accordion` that aligns at the `end`. 

By adding an option for users to specify alignment with a prop, there's no need for them to add style overrides like we (originally) did. 

## Available extra resources

I can contribute this update. 👍

### PR DESCRIPTION
**Pull request by jendowns**
Closes #4584

Proposal to add an `align` prop to the `Accordion` and `AccordionSkeleton`.

By default, `align="end"`, which is the current Carbon default.
If `align="start"`, then the heading text + arrow order is reversed, with the arrow being at the "start" of the heading row.

#### Changelog

**New**

- add `align` prop with class name and styles

**Changed**

- update storybook to include `align` prop knobs
- update snapshots

#### Testing / Reviewing

In the `carbon-components-react` deployment for this PR, open the knobs & use the `align` knob to change alignment of the `Accordion` and `AccordionSkeleton` headings.

### CODE REVIEW

File: packages/components/src/components/accordion/_accordion.scss [hunk 1]
```diff
@@ -23,6 +23,22 @@
     width: 100%;
   }
 
+  .#{$prefix}--accordion--start {
+    .#{$prefix}--accordion__heading {
```
Comment (joshblack) [inline packages/components/src/components/accordion/_accordion.scss]: Could you un-nest selectors here?

**Review by joshblack — CHANGES_REQUESTED** [review]

File: packages/components/src/components/accordion/_accordion.scss (hunk 1 above)
Comment (jendowns) [reply packages/components/src/components/accordion/_accordion.scss]: Thanks @joshblack! Just pushed an update that unnests these selectors 👍

**Comment by joshblack** [comment]
@carbon-design-system/design was curious if this was functionality we wanted to add in 👀

**Comment by aagonzales** [comment]
I think if it's an easy to add and switch between the two then its ok with design. We'll need to maybe think of some guidance on how the two should be used and if there's a usage difference or if its just an aesthetic chose a product can make and then consistently use.

**Review by joshblack — APPROVED** [review]

**Comment by jendowns** [comment]
Thank you @aagonzales! As an example, we have a component in the IBM Security portoflio called a `FilterPanel` that uses inner `Accordion`s with "start" alignment applied: https://ibm-security.carbondesignsystem.com/?path=/story/patterns-filterpanel--default

Screen Shot 2019-12-02 at 1 48 18 PM

I'm sure @cameroncalder has more examples / justifications he can share in Slack though! There are probably other patterns out there as well, aside from the component above.

**Review by aledavila — APPROVED** [review]
lgtm

**Review by jillianhowarth — APPROVED** [review]

**Comment by joshblack** [comment]
@dakahn just wanted to double-check, would this pattern work for the screenshot over in: https://github.com/carbon-design-system/carbon/pull/4754#issuecomment-560553322 ?

**Pull request merged**